packages/adapter/addon/rest.js

```diff
@@ -309,18 +310,47 @@ const RESTAdapter = Adapter.extend(BuildURLMixin, {
     },
   }),
 
-  useFetch: computed(function() {
-    let ENV = getOwner(this).resolveRegistration('config:environment');
-    // TODO: https://github.com/emberjs/data/issues/6093
-    let jQueryIntegrationDisabled = ENV && ENV.EmberENV && ENV.EmberENV._JQUERY_INTEGRATION === false;
-
-    if (jQueryIntegrationDisabled) {
-      return true;
-    } else if (hasNajax || hasJQuery) {
-      return false;
-    } else {
-      return true;
-    }
+  useFetch: computed({
```

Comment:
Lets move this down into the `if (DEPRECATE_NAJAX) {` conditional, and here (for all users) we'd say `useFetch: true`.

That way users that opt-in to deprecation stripping of `DEPRECATE_NAJAX` deprecation can reduce the overall payload, and we avoid the work all together...